module/users/helper.js

```diff
@@ -467,6 +467,65 @@ module.exports = class UsersHelper {
           });
         }
 
+        let totalCount = 0;
+        let mergedData = [];
+        //fetch all targeted programs to users
+        let targetedPrograms = await programsHelper.forUserRoleAndLocation(
+          requestedData,
+          "", // not passing page size
+          "", // not passing page number
+          "",
+          ["_id"],
+          programId
+        );
+        let targetedProgramIds = gen.utils.convertArrayObjectIdtoStringOfObjectId(targetedPrograms.data);
+
+        // check current program is targeted or not
+        if(targetedProgramIds.length === 0){
+          let solutionIds = []
+          let importedSurveys = await surveyService.userSurveys(token,programId)
```

Comment:
@ankitpws It will be efficient to dall all 3 calls in paralle.

Reply:
yes it will be fine or else we have to promise chaining or promise all both will be not good option.